File: 39-drag-and-drop/csharp/DragAndDrop.cs
using System.Threading;
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;

namespace DragAndDrop
{
    public class DragAndDrop
    {
        IWebDriver Driver;
        IJavaScriptExecutor JSDriver;

        [SetUp]
        public void SetUp()
        {
            Driver = new FirefoxDriver();
            JSDriver = (IJavaScriptExecutor)Driver;
        }

        [TearDown]
        public void TearDown()
        {
            Driver.Quit();
        }

        [Test]
        public void DragAndDropExample()
        {
            Driver.Navigate().GoToUrl("https://the-internet.herokuapp.com/drag_and_drop");

            var source = Driver.FindElement(By.Id("column-a"));
            var target = Driver.FindElement(By.Id("column-b"));
            var dnd_javascript = Properties.Resources.dnd;

            JSDriver.ExecuteScript(dnd_javascript + "$('#column-a').simulateDragDrop({ dropTarget: '#column-b'});");

            Thread.Sleep(1500);
            Assert.That(source.Text, Is.EqualTo("B"), "Source text was not 'B'");
            Assert.That(target.Text, Is.EqualTo("A"), "Target text was not 'A'");
        }
    }
}

File: 39-drag-and-drop/python/drag-and-drop.py
# -*- coding: utf-8 -*-
"""
Implementation of http://elementalselenium.com/tips/39-drag-and-drop
	DragAndDrop is currently not available on selenium anymore without JS injection.
"""

import os
import time
import unittest

from selenium import webdriver
from selenium.webdriver.common.action_chains import ActionChains

class DragAndDrop(unittest.TestCase):

	def setUp(self):
		self.driver = webdriver.Firefox()

	def tearDown(self):
		self.driver.quit()

	def test_example_1(self):
		self.driver.get('http://the-internet.herokuapp.com/drag_and_drop')

		source = self.driver.find_element_by_id('column-a')
		target = self.driver.find_element_by_id('column-b')

		ActionChains(self.driver).drag_and_drop(source, target).perform()

		time.sleep(3)

		assert (self.driver.find_element_by_id('column-a').text == 'B' and self.driver.find_element_by_id('column-b').text == 'A'), "Issue with classic drag-and-drop"

	def test_example_2(self):
		self.driver.get('http://the-internet.herokuapp.com/drag_and_drop')

		with open(os.path.dirname(os.path.abspath(__file__)) + '\dnd.js') as f:
			dndJS = f.read()

		self.driver.execute_script(dndJS + "$('#column-a').simulateDragDrop({dropTarget:'#column-b'});")

		time.sleep(3)

		assert (self.driver.find_element_by_id('column-a').text == 'B' and self.driver.find_element_by_id('column-b').text == 'A'), "Issue with JQuery drag-and-drop"

if __name__ == '__main__':
	unittest.main()